graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -258,11 +288,16 @@ public GraphqlService build() {
             configurer.configure(builder);
         }
 
-        final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
-        final List<Consumer<? super DataLoaderRegistry>> dataLoaderRegistries =
-                dataLoaderRegistryConsumers.build();
-        for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistries) {
-            configurer.accept(dataLoaderRegistry);
+        if (dataLoaderRegistryConsumers != null) {
+            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+            for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistryConsumers.build()) {
+                configurer.accept(dataLoaderRegistry);
+            }
+            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
+        }
+        if (dataLoaderRegistryFactory == null) {
+            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
+            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
         }
```

Comment:
nit: We shouldn't set the field in this build() method.
```suggestion
	final Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory;
        if (this.dataLoaderRegistryFactory != null) {
            dataLoaderRegistryFactory = this.dataLoaderRegistryFactory;
        } else if (dataLoaderRegistryConsumers != null) {
            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
            for (Consumer<? super DataLoaderRegistry> configurer : dataLoaderRegistryConsumers.build()) {
                configurer.accept(dataLoaderRegistry);
            }
            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
        } else {
            final DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
            dataLoaderRegistryFactory = ctx -> dataLoaderRegistry;
        }
```